internal/clusterinfo/clusterinfo.go

```diff
@@ -76,14 +78,31 @@ func Discover(ctx context.Context, kubeClient kubernetes.Interface) (Info, error
 }
 
 // IsManagedKyma returns true if the k8s go-client is configured to access a managed kyma runtime
-func IsManagedKyma(restConfig *rest.Config) bool {
-	//Legacy, may be removed in the future
+func IsManagedKyma(ctx context.Context, restConfig *rest.Config, kubeClient kubernetes.Interface) (bool, error) {
+	//1) Verfiy the domain
 	if strings.HasSuffix(restConfig.Host, legacyEnvAPIServerSuffix) {
-		return true
+		//Legacy, may be removed in the future
+		return lookupConfigMapMarker(ctx, kubeClient)
 	}
 	if strings.HasSuffix(restConfig.Host, skrEnvAPIServerSuffix) {
-		return true
+		return lookupConfigMapMarker(ctx, kubeClient)
 	}
 
-	return false
+	return false, nil
+}
+
+func lookupConfigMapMarker(ctx context.Context, kubeClient kubernetes.Interface) (bool, error) {
+	opts := metav1.ListOptions{LabelSelector: "reconciler.kyma-project.io/managed-by=reconciler"}
+	cmList, err := kubeClient.CoreV1().ConfigMaps("kyma-system").List(ctx, opts)
```

Comment:
this will be a huge list for some clusters and can lead to a very long lookup, is this accepted as tradeoff?

Reply:
Changed to `Get()` instead of `List()`